restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1361,8 +1362,16 @@ public void onConnectivityEvent(String jobId, RCDeviceListener.RCConnectivitySta
    {
       RCLogger.i(TAG, "onConnectivityEvent(): id: " + jobId + ", connectivityStatus: " + connectivityStatus);
       cachedConnectivityStatus = connectivityStatus;
-      if (state == DeviceState.OFFLINE && connectivityStatus != RCDeviceListener.RCConnectivityStatus.RCConnectivityStatusNone) {
-         state = DeviceState.READY;
+
+      storageManagerPreferences = new StorageManagerPreferences(this);
```

Comment:
Can we move these 4 lines inside a (potentially static) method on FCM handler, since push configuration is its responsibility? Like: `boolean hasPushRegistrationFailed()`?

Reply:
I think it will be overkill. lets talk on daily.